File: tests/integration/call_component_failures/python/component.py
# Copyright 2016-2024, Pulumi Corporation.  All rights reserved.

from typing import Optional

import khulnasoft

class Component(khulnasoft.ComponentResource):
    foo: khulnasoft.Output[str]

    def __init__(self, name: str, foo: khulnasoft.Input[str], opts: Optional[khulnasoft.ResourceOptions] = None):
        props = dict()
        props["foo"] = foo
        super().__init__("testcomponent:index:Component", name, props, opts, True)

    def get_message(self) -> khulnasoft.Output[str]:
        __args__ = dict()
        __args__['__self__'] = self
        return khulnasoft.runtime.call('testcomponent:index:Component/getMessage', __args__, res=self)

File: tests/integration/construct_component_output_values/nodejs/component.ts
// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.

import * as khulnasoft from "@khulnasoft/khulnasoft";

export interface BarArgs {
    tags?: khulnasoft.Input<{[key: string]: khulnasoft.Input<string>}>;
}

export interface FooArgs {
    something?: khulnasoft.Input<string>;
}

export interface ComponentArgs {
    bar?: khulnasoft.Input<BarArgs>;
    foo?: FooArgs;
}

export class Component extends khulnasoft.ComponentResource {
    constructor(name: string, args?: ComponentArgs, opts?: khulnasoft.ComponentResourceOptions) {
        super("testcomponent:index:Component", name, args, opts, true /*remote*/);
    }
}
